src/Analysis/Ast/Impl/Analyzer/Caching/CacheFolders.cs

```diff
@@ -40,33 +40,43 @@ internal static class CacheFolders {
         public static string GetCacheFolder(IServiceContainer services) {
             var platform = services.GetService<IOSPlatform>();
             var logger = services.GetService<ILogger>();
-            
+
             // Default. Not ideal on all platforms, but used as a fall back.
             var localAppData = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
             var defaultCachePath = Path.Combine(localAppData, "Microsoft", "Python Language Server");
 
-            string path = null;
+            string cachePath = null;
             try {
                 // %% are used to work around https://github.com/dotnet/corefx/issues/28890
                 const string plsSubfolder = "Microsoft/Python.Language.Server";
-                if (platform.IsMac) {
-                    var expanded = Environment.ExpandEnvironmentVariables($"%HOME%/Library/Caches/{plsSubfolder}");
-                    path = Path.GetFullPath(expanded);
-                } else if(platform.IsLinux) {
-                    var expanded = Environment.ExpandEnvironmentVariables("%XDG_CACHE_HOME%");
-                    if (!string.IsNullOrEmpty(expanded)) {
-                        path = Path.Combine(Path.GetFullPath(expanded), plsSubfolder);
-                    } else {
-                        expanded = Environment.ExpandEnvironmentVariables("%HOME%/.cache");
-                        path = Path.Combine(Path.GetFullPath(expanded), plsSubfolder);
+                var homeFolder = Environment.ExpandEnvironmentVariables("%HOME%");
```

Comment:
`Environment.GetEnvironmentVariable("HOME")`; this will just return `"%HOME%"` if that variable is not set, not null.